astropy/units/tests/test_quantity_array_methods.py

```diff
@@ -640,3 +642,13 @@ def test_assignment_with_different_names(self):
         assert qra[1].value == np.array((-1000.0, -2000.0, -3000.0), qra.dtype)
         # Ensure we do not override dtype names of value.
         assert value.dtype.names == ("x", "y", "z")
+
+
+@pytest.mark.filterwarnings("ignore::UserWarning")
```

Comment:
Do we also want to match the "experimental" warning message, so we don't accidentally ignore an unexpected UserWarning that is unrelated?